pkg/cmd/repo/autolink/list/list_test.go

```diff
@@ -0,0 +1,269 @@
+				autolinks: []autolink{
+					{
+						ID:             1,
+						KeyPrefix:      "TICKET-",
+						URLTemplate:    "https://example.com/TICKET?query=<num>",
+						IsAlphanumeric: true,
+					},
+					{
+						ID:             2,
+						KeyPrefix:      "STORY-",
+						URLTemplate:    "https://example.com/STORY?id=<num>",
+						IsAlphanumeric: false,
+					},
+				},
+			},
+			wantStdout: "[{\"id\":1},{\"id\":2}]\n",
+			wantStderr: "",
+		},
+		{
+			name:  "list non-tty",
+			opts:  &listOptions{},
+			isTTY: false,
+			stubLister: stubAutoLinkLister{
+				autolinks: []autolink{
+					{
+						ID:             1,
+						KeyPrefix:      "TICKET-",
+						URLTemplate:    "https://example.com/TICKET?query=<num>",
+						IsAlphanumeric: true,
+					},
+					{
+						ID:             2,
+						KeyPrefix:      "STORY-",
+						URLTemplate:    "https://example.com/STORY?id=<num>",
+						IsAlphanumeric: false,
+					},
+				},
+			},
+			wantStdout: heredoc.Doc(`
+				1	TICKET-	https://example.com/TICKET?query=<num>	true
+				2	STORY-	https://example.com/STORY?id=<num>	false
+			`),
+			wantStderr: "",
+		},
+		{
+			name:  "no results",
+			opts:  &listOptions{},
+			isTTY: true,
+			stubLister: stubAutoLinkLister{
+				autolinks: []autolink{},
+			},
+			expectedErr: cmdutil.NewNoResultsError("no autolinks found in OWNER/REPO"),
+			wantStderr:  "",
+		},
+		{
+			name:  "client error",
+			opts:  &listOptions{},
+			isTTY: true,
+			stubLister: stubAutoLinkLister{
+				autolinks: []autolink{},
+				err:       testAutolinkClientListError{},
+			},
+			expectedErr: testAutolinkClientListError{},
+			wantStderr:  "",
+		},
+		{
+			name:       "web mode",
+			isTTY:      true,
+			opts:       &listOptions{WebMode: true},
+			wantStderr: "Opening https://github.com/OWNER/REPO/settings/key_links in your browser.\n",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ios, _, stdout, stderr := iostreams.Test()
+			ios.SetStdoutTTY(tt.isTTY)
+			ios.SetStdinTTY(tt.isTTY)
+			ios.SetStderrTTY(tt.isTTY)
+
+			opts := tt.opts
+			opts.IO = ios
+			opts.Browser = &browser.Stub{}
+
+			opts.IO = ios
+			opts.BaseRepo = func() (ghrepo.Interface, error) { return ghrepo.New("OWNER", "REPO"), nil }
+
+			opts.AutolinkClient = &tt.stubLister
+			err := listRun(opts)
+
+			if tt.expectedErr != nil {
+				require.Error(t, err)
+				require.ErrorIs(t, err, tt.expectedErr)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.wantStdout, stdout.String())
+			}
+
+			if tt.wantStderr != "" {
+				assert.Equal(t, tt.wantStderr, stderr.String())
+			}
+		})
+	}
+}
+
+type (
+	testAutolinkClientListError struct{}
+
+	stubAutoLinkLister struct {
+		autolinks []autolink
+		err       error
+	}
+)
+
+func (g stubAutoLinkLister) List(repo ghrepo.Interface) ([]autolink, error) {
+	return g.autolinks, g.err
+}
+
+func (e testAutolinkClientListError) Error() string {
+	return "autolink client list error"
+}
```

Comment:
Small style nit: I see no other instances of types declared in blocked parens like this. I think we should change this to 

```suggestion
type stubAutoLinkLister struct {
	autolinks []autolink
	err       error
}

func (g stubAutoLinkLister) List(repo ghrepo.Interface) ([]autolink, error) {
	return g.autolinks, g.err
}

type testAutolinkClientListError struct{}

func (e testAutolinkClientListError) Error() string {
	return "autolink client list error"
}
```

I think it also might make sense to move these above `TestListRun` for readability, but I don't feel strongly.